drivers/can/can_mcp2515.c

```diff
@@ -606,6 +594,25 @@ static struct mcp2515_data mcp2515_data_1 = {
 	.filter_usage = 0U,
 };
 
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_SJW >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_SJW <= 4),
+	"1 <= SJW <= 4");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PROP_SEG >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_PROP_SEG <= 8),
+	"1 <= PROP <= 8");
+BUILD_ASSERT_MSG((DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG1 >= 1) &&
+	(DT_INST_0_MICROCHIP_MCP2515_PHASE_SEG1 <= 8),
+	"1 <= BS1 <= 8");
```

Comment:
Looking good!  For consistency should BS1, BS2 and PROP be PHASE_SEG1, PHASE_SEG2 and PROP_SEG?

Reply:
Phew to be honest no idea why I named them BS1 and BS2, should be PS1/ps1 and PS2/ps2, weird. I think I mainly wanted short names because the 80 char character limit is tough enough as is with all the comparisons in the checks.